keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -1199,6 +1200,42 @@ class SqlArtifactRepository(
     }
   }
 
+  override fun getGitMetadataByPromotionStatus(
+    deliveryConfig: DeliveryConfig,
+    environmentName: String,
+    artifactReference: String,
+    promotionStatus: String
+  ): GitMetadata? {
+    return sqlRetry.withRetry(READ) {
+
+      val artifact = deliveryConfig.artifacts.firstOrNull { it.reference == artifactReference }
+        ?: error("Artifact not found: name=$artifactReference, deliveryConfig=${deliveryConfig.name}")
+
+      val version = jooq
+        .select(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION)
+        .from(ENVIRONMENT_ARTIFACT_VERSIONS)
+        .where(ENVIRONMENT_ARTIFACT_VERSIONS.ENVIRONMENT_UID.eq(deliveryConfig.getUidFor(environmentName)))
+        .and(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_UID.eq(artifact.uid))
+        .and(ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_STATUS.eq(promotionStatus))
+        .orderBy(ENVIRONMENT_ARTIFACT_VERSIONS.DEPLOYED_AT.desc())
+        .fetch(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION)
+        .firstOrNull()
+
+      jooq
```

Comment:
You should only execute this select here if `version != null`, right?

Reply:
yep, added that check